jenkins/deploy.sh

```diff
@@ -17,76 +17,81 @@
 ###
 
 set -ex
-SIGN_FILE=$1
-DATABRICKS=$2
-VERSIONS_BUILT=$3
 
-export M2DIR=${M2DIR:-"$WORKSPACE/.m2"}
+SIGN_FILE=${1:-"false"}
+DIST_PL=${DIST_PL:-"dist"}
+SQL_PL=${SQL_PL:-"sql-plugin"}
+POM_FILE=${POM_FILE:-`find "$DIST_PL/target/extra-resources/" -name pom.xml`}
+OUT_PATH=${OUT_PATH:-"$DIST_PL/target"}
+SIGN_TOOL=${SIGN_TOOL:-"gpg"}
+MVN_SETTINGS=${MVN_SETTINGS:-"jenkins/settings.xml"}
 
+MVN="mvn -B -Dmaven.wagon.http.retryHandler.count=3 -DretryFailedDeploymentCount=3 -s $MVN_SETTINGS"
 ###### Build the path of jar(s) to be deployed ######
-
-cd $WORKSPACE
-
-###### Databricks built tgz file so we need to untar and deploy from that
-if [ "$DATABRICKS" == true ]; then
-    rm -rf deploy
-    mkdir -p deploy
-    cd deploy
-    tar -zxf ../spark-rapids-built.tgz
-    cd spark-rapids
-fi
-
-ART_ID=`mvn help:evaluate -q -pl $DIST_PL -Dexpression=project.artifactId -DforceStdout`
-ART_VER=`mvn help:evaluate -q -pl $DIST_PL -Dexpression=project.version -DforceStdout`
+ART_ID=`$MVN help:evaluate -q -pl $DIST_PL -Dexpression=project.artifactId -DforceStdout`
+ART_VER=`$MVN help:evaluate -q -f $DIST_PL -Dexpression=project.version -DforceStdout`
 CUDA_CLASSIFIER=`mvn help:evaluate -q -pl $DIST_PL -Dexpression=cuda.version -DforceStdout`
-
-FPATH="$DIST_PL/target/$ART_ID-$ART_VER-$CUDA_CLASSIFIER"
-POM_FPATH="$DIST_PL/target/extra-resources/META-INF/maven/com.nvidia/$ART_ID/pom.xml"
+FPATH="$OUT_PATH/$ART_ID-$ART_VER"
+cp $FPATH-$CUDA_CLASSIFIER.jar $FPATH.jar
+
+# dist does not have javadoc and sources jars, use 'sql-plugin' instead
+source jenkins/version-def.sh >/dev/null 2&>1
+echo $SPARK_BASE_SHIM_VERSION
+SQL_ART_ID=`mvn help:evaluate -q -pl $SQL_PL -Dexpression=project.artifactId -DforceStdout`
+SQL_ART_VER=`mvn help:evaluate -q -pl $SQL_PL -Dexpression=project.version -DforceStdout`
+JS_FPATH="${SQL_PL}/target/spark${SPARK_BASE_SHIM_VERSION}/${SQL_ART_ID}-${SQL_ART_VER}"
+cp $JS_FPATH-sources.jar $FPATH-sources.jar
+cp $JS_FPATH-javadoc.jar $FPATH-javadoc.jar
 
 echo "Plan to deploy ${FPATH}.jar to $SERVER_URL (ID:$SERVER_ID)"
 
-FINAL_AGG_VERSION_TOBUILD=${FINAL_AGG_VERSION_TOBUILD:-'311'}
 ###### Choose the deploy command ######
-MVN="mvn -Dmaven.wagon.http.retryHandler.count=3 -DretryFailedDeploymentCount=3"
 if [ "$SIGN_FILE" == true ]; then
-    # No javadoc and sources jar is generated for shade artifact only. Use 'sql-plugin' instead
-    SQL_ART_ID=`mvn help:evaluate -q -pl $SQL_PL -Dexpression=project.artifactId -DforceStdout`
-    SQL_ART_VER=`mvn help:evaluate -q -pl $SQL_PL -Dexpression=project.version -DforceStdout`
-    JS_FPATH="${SQL_PL}/target/spark${FINAL_AGG_VERSION_TOBUILD}/${SQL_ART_ID}-${SQL_ART_VER}"
-    SRC_DOC_JARS="-Dsources=${JS_FPATH}-sources.jar -Djavadoc=${JS_FPATH}-javadoc.jar"
-    DEPLOY_CMD="$MVN -B gpg:sign-and-deploy-file -s jenkins/settings.xml -Dgpg.passphrase=$GPG_PASSPHRASE"
+    case $SIGN_TOOL in
+        nvsec)
+            cp $NVSEC_CFG_FILE ~/.nvsec.cfg
```

Comment:
This is a bit surprising.  What if the user running this script had their own custom settings in ~/.nvsec.cfg?  Are we sure there's no way to pass an explicit config path to nvsec?

Reply:
What if the user running this script had their own custom settings in ~/.nvsec.cfg? 
     --> The file `.nvsec.cfg` is only for signing jars with the `nvsec` tool, the additional Sonatype credentials to are needed to deploy to OSS staging repo. External user should have no secrets of `.nvsec.cfg`, so they can not sign via `nvsec`

Are we sure there's no way to pass an explicit config path to nvsec? 
    --> There is a way to pass config path to nvsec, but we are wrapping `nvsec` as the `gpg` executable of the plugin `gpg:sign-and-deploy-file`, so we need to PASS the `.nvsec.cfg` path via gpg plugin. This is a workaround and odd way, as GPG plugin need not such file.

Let me discuss with @pxLi  later to see if we have better way te set nvsec credentials, maybe apply nvsec secret from Jenkinsfile is a way.

Reply:
> maybe apply nvsec secret from Jenkinsfile is a way.
yeah, I agree to put the nvsec config setup in pipeline side

Reply:
fixed, moved nvsec config setup into pipeline side